### PR TITLE
Initial collateral types

### PR DESCRIPTION
**Pull request by nieldeckx**
There are 3 TODOs left in the tests for which I expected other values. Can you please check and let me know?

### CODE REVIEW

File: clarity/contracts/arkadiko-collateral-types-v1-1.clar [hunk 1]
```diff
@@ -217,12 +217,12 @@
...
+      liquidation-ratio: u175, ;;175
       collateral-to-debt-ratio: u200,
-      maximum-debt: u100000000000000,
+      maximum-debt: u500000000000000, ;; 500 mio
       liquidation-penalty: u1000, ;; 10% in basis points
-      stability-fee: u95908125, ;; 0.001363077% daily percentage == 0.5% APY, 0.0000094658125% per 10 minutes (1 block)
-      stability-fee-decimals: u15,
+      stability-fee: u7610350076, ;; 4% / 365 days / (24*6) blocks = 0.00007610350076 fee per block
+      stability-fee-decimals: u16,
       stability-fee-apy: u50 ;; 50 basis points
```
Comment (philipdesmedt) [inline clarity/contracts/arkadiko-collateral-types-v1-1.clar:226]: REQ: this should be u400 (4%) - this is used to display in the UI.

File: clarity/contracts/arkadiko-collateral-types-v1-1.clar [hunk 2]
```diff
@@ -235,31 +235,13 @@
...
-      total-debt: u0,
-      liquidation-ratio: u200,
-      collateral-to-debt-ratio: u300,
-      maximum-debt: u10000000000000,
-      liquidation-penalty: u1300,
-      stability-fee: u191816250, ;; 0.002726155% daily percentage == 1% APY
+      stability-fee: u1331811263, ;; 7% / 365 days / (24*6) blocks = 0.0001331811263 fee per block
       stability-fee-decimals: u15,
-      stability-fee-apy: u100
+      stability-fee-apy: u100 ;; 100 basis points
```
Comment (philipdesmedt) [inline clarity/contracts/arkadiko-collateral-types-v1-1.clar:244]: REQ: this should be u700 (7%) - this is used to display in the UI.

File: clarity/tests/arkadiko-auction-engine-v1-1_test.ts [hunk 3]
```diff
@@ -548,7 +548,7 @@ Clarinet.test({
     // Initialize price of STX to $2 in the oracle
     let result = oracleManager.updatePrice("STX", 200);
 
-    // Create vault for wallet_1 - 1000 STX, 1100 xUSD
+    // Create vault for wallet_1 - 1000 STX, 100 xUSD
```
Comment (philipdesmedt) [inline clarity/tests/arkadiko-auction-engine-v1-1_test.ts:551]: REQ: 1100 xUSD, no?

File: clarity/tests/arkadiko-freddie-v1-1_test.ts [hunk 4]
```diff
@@ -79,7 +79,7 @@ Clarinet.test({
     result.expectOk().expectUint(1925000);
 
     let call = vaultManager.getCurrentCollateralToDebtRatio(1, deployer);
-    call.result.expectOk().expectUint(200);
+    call.result.expectOk().expectUint(199);
```
Comment (philipdesmedt) [inline clarity/tests/arkadiko-freddie-v1-1_test.ts:82]: REQ/Q: this might be 199 since the stability fee is still commented out (see TODO in `calculate-current-collateral-to-debt-ratio`).. but probably 200 is correct, not sure

File: clarity/tests/arkadiko-freddie-v1-1_test.ts [hunk 5]
```diff
@@ -341,36 +342,21 @@ Clarinet.test({
...
+        types.principal("STSTW15D618BSZQB85R058DS46THH86YQQY6XCB7.arkadiko-sip10-reserve-v1-1"),
         types.principal("STSTW15D618BSZQB85R058DS46THH86YQQY6XCB7.arkadiko-token"),
       ], deployer.address)
     ]);
     block.receipts[0].result
       .expectErr()
       .expectUint(45);
-    
-    // Mint extra
-    result = vaultManager.mint(deployer, 1, 0.000001);
```
Comment (philipdesmedt) [inline clarity/tests/arkadiko-freddie-v1-1_test.ts:364]: Q: why did you remove these?

**Review by philipdesmedt — CHANGES_REQUESTED** [review]

File: clarity/tests/arkadiko-stacker-v1-1_test.ts [hunk 6]
```diff
@@ -325,11 +325,17 @@ Clarinet.test({
     let [stxTransferEvent1, stxTransferEvent2] = block.receipts[1].events;
     stxTransferEvent1.stx_transfer_event.sender.expectPrincipal("STSTW15D618BSZQB85R058DS46THH86YQQY6XCB7.arkadiko-stacker-v1-1");
     stxTransferEvent1.stx_transfer_event.recipient.expectPrincipal(deployer.address);
-    stxTransferEvent1.stx_transfer_event.amount.expectInt(312498000);
+
+    // TODO - Why did this change? Would have expected it to remain the same
```
Comment (philipdesmedt) [inline clarity/tests/arkadiko-stacker-v1-1_test.ts:329]: @nieldeckx you change the STX from 1000 to 1500 collateral..

since we sold off less collateral, we have more leftover (~543, see line 306). As a result, the liquidators bought less in auctions, and thus are eligible for less STX from PoX pro-rata as well.

You raised the collateral with 1/3rd (1000 -> 1500STX). The dump in price stayed the same ($2 -> $1.5)... so we sell of 1/3rd less collateral as well, and stackers get 1/3rd less collateral (312498000-(312498000/3) = 208332000)

File: clarity/tests/arkadiko-freddie-v1-1_test.ts (hunk 5 above)
Comment (nieldeckx) [reply clarity/tests/arkadiko-freddie-v1-1_test.ts:364]: In the initial test we created a vault with collateral type DIKO-A, and tried to call deposit/withdraw/mint/burn with wrong parameters. (ex: passing STX reserve to withdraw should fail). But DIKO-A has been removed so test not needed for now.

File: clarity/tests/arkadiko-freddie-v1-1_test.ts (hunk 4 above)
Comment (nieldeckx) [reply clarity/tests/arkadiko-freddie-v1-1_test.ts:82]: Yes I think so, but I have no clue how to fix the TODO in Freddie :/

File: clarity/tests/arkadiko-stacker-v1-1_test.ts (hunk 6 above)
Comment (nieldeckx) [reply clarity/tests/arkadiko-stacker-v1-1_test.ts]: Alright thx for the explanation.

**Comment by philipdesmedt** [comment]
@nieldeckx can you fix the merge conflict?
